src/components/implementation/tests/unit_coreid/coreid_test.c

```diff
@@ -0,0 +1,14 @@
+#include <cos_kernel_api.h>
+#include <cos_types.h>
+
+
+int
+parallel_main(coreid_t cid)
+{
+	assert(cid == cos_coreid());
+
+	/* prints need to be serialized for this test to be useful */
```

Comment:
If this is the case, why not depend on the print component so that they are serialized? It would also print out the core id as part of the print component, so you could validate that this print replicates that.